app/src/components/guestbook/GuestBookMessageEdit.tsx

```diff
@@ -34,45 +41,93 @@ export default function GuestBookMessageEdit({
         []
     );
 
-    const updateMessage = useCallback(
-        (newValues: Partial<IGuestBookMessage>) => {
-            setMessage({
-                ...(message as IGuestBookMessage),
-                ...newValues,
-            });
-        },
-        [message, setMessage]
-    );
+
+    const initializeValues = (): NewGuestBookMessage | IGuestBookMessage => {
+        if (isUndefined(message)){
+            return getInitialMessage()
+        }
+        return message!
+    }
+
+    const validationSchema = Yup.object({
+        author: Yup.string()
+            .matches(
+                /^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi,
+                'Name can only contain Latin letters.'
+            )
+            .min(2, 'Too short!')
+            .max(50, 'Too long!')
+            .required('Required!'),
+        message: Yup.string().min(2, 'Too short!').max(1000, 'Too long!').required('Required!'),
+    });
 
     return (
-        <Form>
-            <Form.Group className="mb-3" controlId="EditMessage.Author">
-                <Form.Label>Name</Form.Label>
-                <Form.Control
-                    type="text"
-                    placeholder={placeholderName}
-                    defaultValue={message?.author}
-                    onChange={(event) => updateMessage({ author: event.target.value })}
-                />
-            </Form.Group>
-            <Form.Group className="mb-3" controlId="EditMessage.Message">
-                <Form.Label>Message</Form.Label>
-                <Form.Control
-                    as="textarea"
-                    rows={3}
-                    defaultValue={message?.message}
-                    onChange={(event) => updateMessage({ message: event.target.value })}
-                />
-            </Form.Group>
-            <Stack direction="horizontal" gap={3} className="justify-content-end">
-                <div className="my-auto"></div>
-                <ButtonLink variant="secondary" href={cancelHref}>
-                    Cancel
-                </ButtonLink>
-                <Button variant="primary" onClick={onSubmit} disabled={isSubmitLoading}>
-                    {isSubmitLoading ? <Spinner animation="border" size="sm" /> : submitLabel}
-                </Button>
-            </Stack>
-        </Form>
+        <Formik
+            initialValues={initializeValues()}
+            validationSchema={validationSchema}
+            onSubmit={(message, { setSubmitting }) => {
+                try {
+                    if (onEdit && 'messageId' in message){
```

Comment:
There should be a space between `) {`, you can install prettier to autoformat correctly